handlers/external/external.go

```diff
@@ -344,7 +356,40 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	return status, nil
 }
 
-func replaceVariables(text string, variables map[string]string, contentType string) string {
+func buildQuickRepliesResponse(quickReplies []string, sendMethod string, contentType string) string {
+	if quickReplies == nil {
+		quickReplies = []string{}
+	}
+	if (sendMethod == http.MethodPost || sendMethod == http.MethodPut) && contentType == contentJSON {
+		marshalled, _ := json.Marshal(quickReplies)
+		return string(marshalled)
+	} else if (sendMethod == http.MethodPost || sendMethod == http.MethodPut) && contentType == contentXML {
+		response := bytes.Buffer{}
+
+		for _, reply := range quickReplies {
+			buf := &bytes.Buffer{}
+			err := xml.EscapeText(buf, []byte(reply))
+
+			if err == nil {
+				reply = buf.String()
+				response.WriteString(fmt.Sprintf("<item>%s</item>", reply))
+			}
+		}
+		return response.String()
```

Comment:
Can't you just use `xml.Marshal` here with an appropriately named struct?

Reply:
Do you mean create a struct like the following?

```
type foo struct {
  items []string `xml:item`
}
```

So, the `xml.Marshal` could return something like:

```
"<foo><item>a</item><item>b</item><item>c</item></foo>"
```

Still needed to remove the `foo` tag, alright?

Reply:
I think I'd be alright even with just concatenating the single `<item>foo</item>` elements together from individual `xml.Marshal`s